srml/metadata/src/lib.rs

```diff
@@ -261,7 +261,13 @@ pub enum StorageFunctionType {
 	Map {
 		key: DecodeDifferentStr,
 		value: DecodeDifferentStr,
-	}
+	},
+	DoubleMap {
+		key1: DecodeDifferentStr,
+		key2: DecodeDifferentStr,
+		value: DecodeDifferentStr,
+		key2_hasher: DecodeDifferentStr,
```

Comment:
at first I wanted twox128 for both keys, but this can be dangerous as in some cases like in contracts we don't trust the second key entry and thus they can be build to lower capacity of the trie. (afaik, to other trie entries with same starting part).
So contract use blake2_256 for this reason. So I made this configurable as user want to choose depending and trustness of second key.

I'm not very aware of trie implementation, you mean key lenght must less than 32bytes ? concat(key1, blake2_256(key2)) is 48 bytes.

Reply:
I you check https://github.com/cheme/substrate/blob/master/core/trie/src/node_header.rs the key length and other variant are encoded on a single byte: making LEAF_NODE_THRESHOLD var 253 -128 = 125 so max key lenghth in such trie seems to be 125 nible (a half byte) so 62 byte (two blake would be 64).